src/main/java/com/ibm/watson/developer_cloud/spring/boot/WatsonAutoConfiguration.java

```diff
@@ -0,0 +1,222 @@
+    return service;
+  }
+
+  // Watson NaturalLanguageUnderstanding service
+
+  @Autowired
+  private WatsonNaturalLanguageUnderstandingConfigurationProperties nluConfig;
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnWatsonServiceProperties(prefix = WatsonNaturalLanguageUnderstandingConfigurationProperties.PREFIX)
+  public NaturalLanguageUnderstanding naturalLanguageUnderstanding() {
+    NaturalLanguageUnderstanding service = new NaturalLanguageUnderstanding(nluConfig.getVersionDate());
+    configUrl(service, nluConfig);
+    configBasicAuth(service, nluConfig);
+    return service;
+  }
+
+  // Watson PersonalityInsights service
+
+  @Autowired
+  private WatsonPersonalityInsightsConfigurationProperties piConfig;
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnWatsonServiceProperties(prefix = WatsonPersonalityInsightsConfigurationProperties.PREFIX)
+  public PersonalityInsights personalityInsights() {
+    PersonalityInsights service = new PersonalityInsights(piConfig.getVersionDate());
+    configUrl(service, piConfig);
+    configBasicAuth(service, piConfig);
+    return service;
+  }
+
+  // Watson SpeechToText service
+
+  @Autowired
+  private WatsonSpeechToTextConfigurationProperties sttConfig;
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnWatsonServiceProperties(prefix = WatsonSpeechToTextConfigurationProperties.PREFIX)
+  public SpeechToText speechToText() {
+    SpeechToText service = new SpeechToText();
+    configUrl(service, sttConfig);
+    configBasicAuth(service, sttConfig);
+    return service;
+  }
+
+  // Watson TextToSpeech service
+
+  @Autowired
+  private WatsonTextToSpeechConfigurationProperties ttsConfig;
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnWatsonServiceProperties(prefix = WatsonTextToSpeechConfigurationProperties.PREFIX)
+  public TextToSpeech textToSpeech() {
+    TextToSpeech service = new TextToSpeech();
+    configUrl(service, ttsConfig);
+    configBasicAuth(service, ttsConfig);
+    return service;
+  }
+
+  // Watson ToneAnalyzer service
+
+  @Autowired
+  private WatsonToneAnalyzerConfigurationProperties taConfig;
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnWatsonServiceProperties(prefix = WatsonToneAnalyzerConfigurationProperties.PREFIX)
+  public ToneAnalyzer toneAnalyzer() {
+    ToneAnalyzer service = new ToneAnalyzer(taConfig.getVersionDate());
+    configUrl(service, taConfig);
+    configBasicAuth(service, taConfig);
+    return service;
+  }
+
+  // Watson VisualRecognition service
+
+  @Autowired
+  private WatsonVisualRecognitionConfigurationProperties vrConfig;
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnWatsonServiceProperties(prefix = WatsonVisualRecognitionConfigurationProperties.PREFIX)
+  public VisualRecognition visualRecognition() {
+    VisualRecognition service = new VisualRecognition(vrConfig.getVersionDate());
+    configUrl(service, vrConfig);
+    configApiKey(service, vrConfig);
```

Comment:
With dedicated VR instances, a username and password is used as opposed to an API key. Do you think we need to add some extra logic in here just in case the user needs that?

Reply:
Good point.  I _think_ we can just add a call to `configBasicAuth`, so that if username/password is specified in application.properties, these will be picked up.  I'll do that.